### DISCUSSION TITLE
Commit Changes After Auto-Increment Build Number

### DISCUSSION THREAD

**Discussion by imjamescrain** [codemagic-ci-cd/codemagic-docs#2446] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Trying to understand how to commit the files changed when Codemagic auto-increments my build number. As I understand it, this will trigger another build since the repo will have changed. Unfortunately, I haven't found any info on this.

### Expected results

Ability to commit changes made after Codemagic auto-increments the build number.

### Actual results

The build is updated, but the code is not committed.

**Comment by himesh-cm** [comment]
Hi @imjamescrain, You don't need to commit to files after incrementing build numbers. Please check our documentation [here](https://docs.codemagic.io/knowledge-codemagic/build-versioning/).
Let me know, if you still have any queries.

**Reply by imjamescrain** [reply]
Thanks @himesh-cm. I re-read the article, but I don't see anything about files being committed to GitHub after they are changed. Are you saying that we _don't need to commit the files_ because we will just be incrementing based off of the build numbers/versions live in the App and Play Store?

**Reply by himesh-cm** [reply]
To provide further clarification, the details of the build versioning are passed as command line arguments to the build command. This means that there is no need to change any files to specify the version details. It is possible that the version details you see locally may be different from those uploaded on app stores. However, since you use these arguments during the build command, the version details are automatically handled during the build process.

**Comment by himesh-cm** [comment]
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
